scalatest/src/main/scala/org/scalatest/prop/PropertyCheckResult.scala

```diff
@@ -15,14 +15,72 @@
+    * good as the property you have defined. But if the property accurately describes the problem, this usually
+    * signifies that things look good.
+    *
+    * @param args The arguments passed in to this check.
+    * @param initSeed The random seed used for this check. This seed is used by the checking code, to pass in to
+    *                 calls to [[Randomizer]]. Reusing the same seed for subsequent runs should produce the same
+    *                 results, so this can be valuable debugging information if you are seeing inconsistent results --
+    *                 this tells you a seed that passed the check.
+    */
   case class Success(args: List[PropertyArgument], initSeed: Long) extends PropertyCheckResult
 
+  /**
+    * This Property Check was inconclusive -- after running for the desired number of times, so many attempts were
+    * discarded that the property as defined should be considered suspect.
+    *
+    * Note that this does *not* mean that the check failed. No falsifying examples were found. The problem is that
+    * the property rejected so many of the generated parameters that it looks a bit iffy.
+    *
+    * TODO: describe how constraints work in this check environment, and precisely how discarding works.
+    *
+    * @param succeeded How many generated values succeeded.
+    * @param discarded How many generated values were rejected as not satisfying the property's preconditions.
+    * @param names TODO: how does this parameter different from argsPassed?
+    * @param argsPassed The arguments passed in to this check.
+    * @param initSeed The random seed used for this check. This seed is used by the checking code, to pass in to
+    *                 calls to [[Randomizer]]. Reusing the same seed for subsequent runs should produce the same
+    *                 results, so this can be valuable debugging information if you are seeing inconsistent results --
+    *                 this tells you a seed that produced indeterminate results.
+    */
   case class Exhausted(succeeded: Long, discarded: Long, names: List[String], argsPassed: List[PropertyArgument], initSeed: Long) extends PropertyCheckResult
 
+  /**
+    * This Property Check did not succeed -- the property was falsified.
+    *
+    * This generally indicates a bug in your code. Values were generated that turned out to produce an error.
+    * The next step is usually to examine these values, and understand why they caused an error.
+    *
+    * Note that this does not ``necessarily`` mean that the code under test is in error: it is possible that the
+    * property itself is incorrect. It is good practice to check both, and figure out which one is wrong.
+    *
+    * @param succeeded How many generated values passed the property before the failure.
+    * @param ex The Exception that was thrown by the property, if any.
+    * @param names TODO: how does this parameter differ from argsPassed?
```

Comment:
I don't remember the difference between names and argsPassed.label. I suspect I was trying to make the error message nice by letting users see the variable name in the source code in the error message itself. That would require a macro, and I'm not sure we are doing that. So we'll need to do more digging to figure that out.

Reply:
Okay, cool.  For now, I'm leaving the TODO as a string around my finger, to remember to keep an eye out for this as I'm digging around in the rest of the code.  (I did a quick scan, but didn't get far enough yet to be clear about usage -- the call stacks are pretty complex.)